src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,199 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { datePickerStyles } from './DatePicker.style';
+import useTheme from 'hooks/useTheme';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import { DateUtils, RangeModifier, DayPickerInputProps } from 'react-day-picker';
+import 'react-day-picker/lib/style.css';
+import dayjs from 'dayjs';
+
+import TextField from '../TextField';
+import Icon from '../Icon';
+import { wrapperStyle } from '../TextField/TextField.style';
+import { flex } from '../../theme/functions';
+import OverlayComponent from './OverlayComponent/OverlayComponent';
+
+export type Props = {
+  /** This boolean shows if the date picker will have the future dates available to select. Default: false */
+  disableFutureDates?: boolean;
+  /** This property is to define if this is a day picker or a day range picker */
+  isRangePicker?: boolean;
+  /** A callback to return user selection */
+  onChange?: (date: DateRange) => DateRange;
+};
+
+export type DateRange =
+  | RangeModifier
+  | {
+      from: undefined;
+      to: undefined;
+    };
+
+export type ExtraOption = { value: string; label: string; dates: Date | Date[] };
+
+const extraOptions: ExtraOption[] = [
+  {
+    value: 'last-7-days',
+    label: 'Last 7 days',
+    dates: [
+      dayjs()
+        .subtract(7, 'day')
+        .toDate(),
+      dayjs().toDate(),
+    ],
+  },
+  {
+    value: 'last-30-days',
+    label: 'Last 30 days',
+    dates: [
+      dayjs()
+        .subtract(30, 'day')
+        .toDate(),
+      dayjs().toDate(),
+    ],
+  },
+
+  {
+    value: 'custom',
+    label: 'Custom',
+    dates: dayjs().toDate(),
+  },
+];
+
+const DatePicker: React.FC<Props> = ({
+  disableFutureDates = false,
+  isRangePicker = false,
+  onChange,
+}) => {
+  const theme = useTheme();
+  const dayPickerRef = useRef<DayPickerInput>(null);
+  const daysInitialState = { from: undefined, to: undefined };
+  const [selectedOption, setSelectedOption] = useState();
+  const [selectedDay, setSelectedDay] = useState<DateRange>({ from: undefined, to: undefined });
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(selectedDay);
+    }
+  }, [onChange, selectedDay]);
+
+  const handleDayRangeClick = useCallback(
+    day => {
+      const aboutToCompleteBothDates = selectedDay.from && !selectedDay.to;
+      const range = DateUtils.addDayToRange(
+        day,
+        // @TODO problem with the library as they define RangeModifiers as not null or undefined when they use it like this
```

Comment:
>they define RangeModifiers as not null or undefined when they use it like this

Is it possible to adhere by their rule?

Reply:
no because in their documentation in order to start the date range you must define it as undefined if you don't need a value at start. Their type has always a date, they don't accept ```undefined``` or ```null```